ui/src/components/helpers/TipTap/BibleVerseExtension.js

```diff
@@ -0,0 +1,93 @@
+import {
+  Node,
+  mergeAttributes,
+  nodeInputRule,
+  nodePasteRule,
+} from '@tiptap/core';
+import { NodeViewWrapper, ReactNodeViewRenderer } from '@tiptap/react';
+import { BibleVerseAccordion } from '../components/BibleVerseAccordion.js';
+import React from 'react';
+
+export const BibleVerseNode = Node.create({
+  name: 'bibleVerse',
+  inline: true,
+  group: 'inline',
+
+  addAttributes() {
+    return {
+      bibleVerse: {
+        default: null,
+      },
+      actionText: {
+        default: null,
+      },
+    };
+  },
+  addCommands() {
+    return {
+      insertBibleVerse:
+        (bibleVerse) =>
+        ({ commands, editor }) => {
+          const position = editor.state.selection.from;
+          return commands.insertContentAt(position, {
+            type: 'bibleVerse',
+            attrs: { bibleVerse },
+          });
+        },
+    };
+  },
+  addInputRules() {
+    return [
+      nodeInputRule({
+        find: /\b((?:READ|TEXT:)[\s\S]*?)\n/,
+        type: this.type,
+        getAttributes: (match) => {
+          return {
+            bibleVerse: match[1].replace('READ ', '').replace('TEXT: ', ''),
+            actionText: match[1].split(' ')[0],
+          };
+        },
+      }),
+    ];
+  },
+  addPasteRules() {
+    return [
+      nodePasteRule({
+        find: /\b((?:READ|TEXT:)[\s\S]*)/g,
+        type: this.type,
+        getAttributes: (match) => {
+          return {
+            bibleVerse: match[1].replace('READ ', '').replace('TEXT: ', ''),
+            actionText: match[1].split(' ')[0],
+          };
+        },
+      }),
+    ];
+  },
+  parseHTML() {
+    return [
+      {
+        tag: 'bible-verse',
+      },
+    ];
+  },
+
+  renderHTML({ HTMLAttributes }) {
+    return ['bible-verse', mergeAttributes(HTMLAttributes)];
```

Comment:
I don't think you need to put mergeAttributes here cuz you aren't merging any attribute from how you are using it